pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -546,3 +549,19 @@ func validateGraph(tasks []PipelineTask) *apis.FieldError {
 	}
 	return nil
 }
+
+func validateMatrix(ctx context.Context, tasks []PipelineTask) (errs *apis.FieldError) {
+	for idx, task := range tasks {
+		errs = errs.Also(task.validateMatrix(ctx).ViaIndex(idx))
+	}
+	return errs
+}
+
+func (pt *PipelineTask) validateMatrix(ctx context.Context) (errs *apis.FieldError) {
```

Comment:
NIT: the pipelineTask member functions are created in `pipeline_types.go` along with its definition:

https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1beta1/pipeline_types.go#L208

We can move this as a followup PR.

Reply:
nice! moved it, thanks :)